### DISCUSSION TITLE
Loving the PID!  Now running on the TB6612 : )

### DISCUSSION THREAD

**Discussion by LukeThomasDawson** [tttapa/Control-Surface-Motor-Fader#14] [General]
G'day Pieter and fellow enthusiasts from Australia,

Luke is me - a huge fan of the Control Surface work you have put together, and an even bigger fan of your patience with the community as you respectfully answer all manner of beginner questions : )  Even the lovable characters out there who have never touched a C++ book in their life and ask things like "but how to I build my program please?"...

Anyhow, a quick note to say I've recently put together the classic motorised fader combo using the good ol' 328P and naturally I came across your projects and dived in to the theory with glee. 

I also used a couple of the TB6612 motor driver's in the form of the DFRobot offering ([these ones](https://wiki.dfrobot.com/2x1.2A_DC_Motor_Driver__TB6612FNG__SKU__DRI0044)) which use an inverter on the direction pin.  This means a standard PWM will drive both forward and reverse - so I modified the PWM timers so that there is only the non-inverting pulse for both forward and backward:
`Clear OC0B on Compare Match, set OC0B at BOTTOM (non-inverting mode)` - do this for both forward and reverse, instead of setting `OC0B` on Compare Match.

...and the results are wonderful.  Clear crisp PID graphs that match up exactly with your control tuning plots.

Happy Midi Motoring out there people.

Best,
Luke

**Comment by tttapa** [comment]
Thanks for the kind words, I'm glad you found the project useful! :)
